activesupport/lib/active_support/testing/filter_chain.rb

```diff
@@ -0,0 +1,51 @@
+module ActiveSupport
+  module Testing
+    class FilterChain # :nodoc:
+      def initialize(runnable, filter, patterns)
+        @runnable = runnable
+        @filters = [ derive_regexp(filter), *derive_filters(patterns) ].compact
+      end
+
+      def any?
+        @filters.any?
+      end
+
+      def ===(method)
+        @filters.any? { |filter| filter === method }
+      end
+
+      private
+        def derive_regexp(filter)
+          filter =~ %r%/(.*)/% ? Regexp.new($1) : filter
```

Comment:
If we didn't duplicate some logic from Minitest here, passing a `-n /hello/` filter won't filter correctly.